### PR TITLE
Disable interactive prompts in CI

### LINKED ISSUE #1357: Skip interactive prompts if `CI` environment variable is set
This is a consistent variable that various CI environments define for the purpose of tooling to change its defaults to make more sense in a noninteractive context.

- https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
- https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables
- https://www.appveyor.com/docs/environment-variables/

The readme calls out that `cargo miri setup` avoids the prompts but it shouldn't be necessary; CI should be able to go directly to `cargo miri test`.

### PR DESCRIPTION
**Pull request by dtolnay**
Closes #1357.

### CODE REVIEW

**Comment by RalfJung** [comment]
Thanks a lot!
Could you also update the README to remove the now-unnecessary `cargo miri setup` from the recommended CI script?

**Comment by dtolnay** [comment]
Good call, done.

**Comment by RalfJung** [comment]
Awesome. :)
@bors r+

**Comment by RalfJung** [comment]
Ah, `setup` uses `ask_user == false` as a signal to print where it put the sysroot... which it rather shouldn't do when the reason for `!ask_user` is that the env var is set. It should only print with `cargo miri setup`.

Honestly we can probably remove that printing entirely, that's from back when I still hard-coded the path in CI config and other places; now we only do that on Windows (due to a lack of batch scripting knowledge) and we have `cargo miri setup --print-sysroot`.

**Comment by dtolnay** [comment]
I pushed 2584507ce2eb9a7fc4c00213d216fc57c794e141 which tries to decouple the various prints from the environment variable check.

File: src/bin/cargo-miri.rs [hunk 1]
```diff
@@ -246,7 +247,9 @@ fn xargo_version() -> Option<(u32, u32, u32)> {
     Some((major, minor, patch))
 }
 
-fn ask_to_run(mut cmd: Command, ask: bool, text: &str) {
```
Comment (RalfJung) [inline src/bin/cargo-miri.rs:249]: I don't think we should forward the `MiriCommand` all the way to this helper function...  but I do agree with the idea of checking the env var in here, that in fact seems like the much better place than where you originally had that check.

Comment (dtolnay) [reply src/bin/cargo-miri.rs:249]: Fixed.

**Comment by RalfJung** [comment]
Thanks! @bors r+

**Pull request merged**